### ISSUE TITLE
Modifying calmcode to SPA ?

### ISSUE THREAD

**Issue by ceroloy** [koaning/calmcode-feedback#23]
Hello,
Would you consider turning calmcode to a SPA? Given the simple and clean design of the website it is easily achievable and I think would it scale much better when you grow the amount of content.

**Comment by koaning** [comment]
Gotta ask .... what is a SPA :)? It deserves mentioning that the goal of the project is "calm", not growth per se.

**Comment by ceroloy** [comment]
@koaning  A SPA is a single page application, where instead of sending html per page(or view) you basically show and hide components made up of html and css. For the routing part and some other details resulting from such organization often SPA frameworks are used (react, vue etc)

> ... not growth per se.

By growth I meant the content really, assuming you are writing html's each time. It might get hard to maintain at some point, but then you mention..

> the goal of the project is "calm"

Maybe we have differing views on what it actually means :) I personally like a bit of abstraction to make my life easy when it comes to maintenance.

**Comment by koaning** [comment]
Now I see :) 

The stack I use is relatively simple. It's all statically generated by jinja. The repository is just a bunch of svg/yml files that generate the entire website using a for-loop. In this sense, it's all pretty much generated and there's barely any javascript involved. I'm less javascript-savvy so I figured keeping it all static is an easy thing to do now. Feel free to correct me if I'm missing out on something but as far as maintainability goes ... it seems hard to beat a few md/yml files.

I've been looking at frontend frameworks to see if it makes sense to make it fully "single-page" and couldn't find one that I really really liked. I figured doing this "generate all files statically" in the meantime is good enough. It's a breeze to maintain and really easy to refactor should I find a frontend framework that I am willing to commit to. People also tell me that there are some SEO benefits of this approach but that's secondary to the simplicity that I allready have.
